DirectProgramming/DPC++FPGA/Tutorials/Features/mem_channels/src/mem_channels.cpp

```diff
@@ -0,0 +1,152 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+#include <chrono>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+
+constexpr size_t vector_size = 1000000; // size of input vectors
+constexpr double kNs = 1e9;             // number of nanoseconds in a second
+
+// Forward declare the kernel name in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class VecAdd;
+
+event runVecAdd(sycl::queue &q, const std::vector<int> &a_vec,
+                const std::vector<int> &b_vec, const std::vector<int> &c_vec,
+                std::vector<int> &sum_vec) {
+#if defined(NO_INTERLEAVING)
```

Comment:
Nothing to do with your tutorial, but I dislike how C++ doesn't have a alternate way of removing code at compile time like this instead of this pre-processor directive.